c7n/resources/asg.py

```diff
@@ -526,21 +526,32 @@ def get_unencrypted_configs(self, ec2):
         return unencrypted_configs
 
     def get_snapshots(self, ec2, snap_ids):
-        """get snapshots corresponding to id, but tolerant of missing."""
+        """get snapshots corresponding to id, but tolerant of invalid id's."""
         while True:
             try:
                 result = ec2.describe_snapshots(SnapshotIds=snap_ids)
             except ClientError as e:
-                if e.response['Error']['Code'] == 'InvalidSnapshot.NotFound':
-                    msg = e.response['Error']['Message']
-                    e_snap_id = msg[msg.find("'") + 1:msg.rfind("'")]
-                    self.log.warning("Snapshot not found %s" % e_snap_id)
-                    snap_ids.remove(e_snap_id)
+                bad_snap = self.get_bad_snapshot(e)
+                if bad_snap:
+                    snap_ids.remove(bad_snap)
                     continue
                 raise
             else:
                 return result.get('Snapshots', ())
 
+    def get_bad_snapshot(self, e):
```

Comment:
this could be a static function which should also make testing easier.

Reply:
Yeah, I was thinking of moving it into utils